### ISSUE TITLE
Report Structure and Content Question

### ISSUE THREAD

**Issue by rossmillward94** [RUG-IDS/Course-2018#13]
Hi,

Regarding the written report part of the assignment, it is mentioned to create a brief written report using either the latex template provided or markdown. The instructions given in the latex template make it quite clear that we are at risk of losing marks if we stray from the specified formatting, is this something we should be considering if opting to generate our report in markdown? Or are we more free regarding layout if we choose to use markdown?

I'm also hoping for clarification on what is expected in the report. Most of the exercises included in the assignment are quite rigid instructions on what to implement, leaving little room for discussion in the report. Are we expected to discuss each exercise individually using the structure provided in the template? 

Thanks, Ross

**Comment by sreejita-rug** [comment]
Dear Ross,
1) In case you are using Markdown please keep the format as close to the given Latex template. Ofcourse it will not be identical and we understand that.
2) In the implementation part of the assignment it is expected that an explanation is provided. For the given assignment, 1.2 has 4 subparts. The organization of report expected is as follows:
   Introduction: 1.2.1 Explain briefly  what F-statistic is, 1.2.2 What PCA is (briefly), and so on on 1.2.3 and 1.2.4
Methodology: 1.2.1 How you implemented the code myOneWayANOVA() - using which mathematical functions and properties, and based on which formula, etc.
1.2.2 How you implemented mypca(): using which mathematical properties 
1.2.3 idem 
1.2.4 idem
Results: We should be able to execute your codes for this. You may also test your functions on a toy dataset
Discussion: Interpret the results obtained in each of the subparts 1.2.1-1.2.4 in separate subsubsections as done for Introduction, and Methods. The discussion part is meant for you to be able to interpret your results.

**Comment by rossmillward94** [comment]
@sreejita-rug Thank you for the response - sounds good 👍

**Comment by JSchilperoort** [comment]
@sreejita-rug You've mentioned that in the results section you should be able to execute our codes. Does anything have to be added to the report in this part? Is it maybe required to actually test the functions on a toy dataset?

**Comment by sreejita-rug** [comment]
Hello @JSchilperoort ,
Testing one's code on a dataset is always a good practise. Infact 1.2.4 asks that the functions are tested on the given dataset. And if you read closely, myEigenFaces, myFeatureSelectionwithGA, and myFeatureSelectionwithANOVA use the some of the functions and concepts which you had to code in 1.2.1-1.2.3. So when you do have to test the subparts in 1.2.4 on the given dataset (as mentioned in 1.2.4) it is a good idea to test the building block functions myOneWayANOVA(), mypca(), etc., as well.

Also, when you are testing your code on a dataset you get a good feedback just by looking at the results, whether your code is doing what it is supposed to do. For 1.2.1-1.2.3 you can use the dataset provided to you for 1.2.4 (face recognition dataset) to also test your codes. Alternatively you can also use any of the datasets from UCI Machine Learning repository, or from Kaggle and inform us in the methods section which one you used, so that we can cross check. 

In the Results section in addition to mentioning that you have submitted the codes myOneWayANOVA(), etc., you can add bar plots* showing the anova on the dataset you selected (for 1.2.1), the principal components (1.2.2), and so on for 1.2.3 . Question 1.2.4 explicitly asks for a comparison between the techniques in myEigenFaces, myFeatureSelectionwithGA, and myFeatureSelectionwithANOVA, using the dataset which has been provided for this.  
Think of the best ways to compare these three functions and explain the evaluation measures you used in the comparison. *[Don't have to limit yourselves to bar plots. ]

Then interpret your findings in the Discussions section. Infact 1.2.4 asks explicitly for a comparison which means, the results obtained from myEigenFaces, myFeatureSelectionwithGA, and myFeatureSelectionwithANOVA, applied on the given dataset, which you show in Results section, have to be discussed- the pros and cons of each function (this is mentioned in the assignment), based on the figures you have used in the Results section. Therefore choose the means of comparison in your Results section well, because you will be forming your arguments in Discussion and Conclusion based on these. 

In conclusion you are expected to add a take home message: which of these techniques would you suggest for which application and why, or which one do you think is a good choice no matter how the application is, and why.

**Comment by liamwietzorrek** [comment]
@sreejita-rug  Does the section '1.1 Exercises:' with calculcations of eigenvaules, eigenvectors and ANOVA by hand also have to be put in the report format (i.e. intro, methods, results, conclusion, discussion)? 
Above you only mention from 1.2.x onwards: 
"In the implementation part of the assignment it is expected that an explanation is provided. For the given assignment, 1.2 has 4 subparts. The organization of report expected is as follows:..."

**Comment by sreejita-rug** [comment]
Hello @liamwietzorrek , 
Only the implementation part has to be in the structure mentioned. The part with calculation, ie., 1.1 would be difficult to be arranged in the aformentioned format. Therefore for that we just expect that you provide the correct number tags, and explain the steps briefly but clearly to show your calculations. And by this I mean that you do not necessarily have to show each and every step, but the main ones. There is another issue about this part where one of your classmates have answered this question. I'll redirect you to his answer https://github.com/RUG-IDS/Course-2018/issues/19#issuecomment-421730751
